scripts/zkai_attention_kv_proof_route_selector_gate.py

```
@@ -3035,6 +3256,87 @@ def validate_d16_quantized_softmax_receipt(summary: Any) -> None:
+    if "no real-valued Softmax" not in summary.get("table_error_bound_policy", ""):
+        raise AttentionKvRouteSelectorError("d16 two-head quantized Softmax real-valued error-bound overclaim")
+    for key in (
+        "source_statement_commitment",
+        "source_public_instance_commitment",
+        "source_score_row_commitment",
+        "source_outputs_commitment",
+        "source_final_kv_cache_commitment",
+    ):
+        commitment = summary.get(key)
+        if not isinstance(commitment, str) or not commitment.startswith("blake2b-256:"):
+            raise AttentionKvRouteSelectorError(f"d16 two-head quantized Softmax {key} drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)